lib/AST/ASTMangler.cpp

```diff
@@ -735,7 +735,9 @@ void ASTMangler::appendOpaqueDeclName(const OpaqueTypeDecl *opaqueDecl) {
   if (canSymbolicReference(opaqueDecl)) {
     appendSymbolicReference(opaqueDecl);
   } else if (auto namingDecl = opaqueDecl->getNamingDecl()) {
+    CanGenericSignature savedSignature = CurGenericSignature;
```

Comment:
Should CurGenericSignature be a stack then, with an RAII utility to save and restore the signature?

Reply:
`llvm::SaveAndRestore` is a good way to indicate what's happening here

Reply:
https://github.com/apple/swift/pull/27093